kernel/src/engine/default/filesystem.rs

```diff
@@ -197,11 +211,8 @@ mod tests {
         let mut url = Url::from_directory_path(tmp.path()).unwrap();
 
         let store = Arc::new(LocalFileSystem::new());
-        let storage = ObjectStoreStorageHandler::new(
-            store,
-            false, // don't have ordered listing
-            Arc::new(TokioBackgroundExecutor::new()),
-        );
+        let storage =
+            ObjectStoreStorageHandler::new(store, Arc::new(TokioBackgroundExecutor::new()));
```

Comment:
aside: Given that we already factored out `store` (presumably for readability), why not also factor out `executor`?
```suggestion
        let executor = Arc::new(TokioBackgroundExecutor::new());
        let storage = ObjectStoreStorageHandler::new(store, executor);
```
(several more below)